app/ide-desktop/lib/dashboard/src/pages/dashboard/Dashboard.tsx

```diff
@@ -122,7 +125,12 @@ export default function Dashboard(props: DashboardProps) {
   const inputBindings = inputBindingsProvider.useInputBindings()
   const [initialized, setInitialized] = React.useState(false)
   const [isHelpChatOpen, setIsHelpChatOpen] = React.useState(false)
-  const [page, setPage] = React.useState(() => localStorage.get('page') ?? pageSwitcher.Page.drive)
+
+  // These pages MUST be ROUTER PAGES.
```

Comment:
are these router pages?

Reply:
Something's wrong?

Reply:
yeah - idk what the comment means

Reply:
Instead of storing the current page in the state we should use Routes for that

Reply:
we can't - the other pages *must* be loaded:
- the editor (GUI) page MUST be loaded (if a project is open), otherwise we have to reconnect (and re-handshake etc, and potentially send MBs of visualization data) to the websockets
- the drive page MUST be loaded as well, to preserve the state of which folders are open
  - also to avoid having to load the folder contents every time the user switches back. for users that are far away from the servers, listing the directory takes a long time (~1 second). you can imagine that having to load multiple folders would make it take even longer
- the home page doesn't have any important special state. (the other two do though, we still need to be able to switch to the home page and back)

Reply:
> the editor (GUI) page MUST be loaded (if a project is open), otherwise we have to reconnect (and re-handshake etc, and potentially send MBs of visualization data) to the websockets

there are better solutions for that. 

> the drive page MUST be loaded as well, to preserve the state of which folders are open

Same here.

> the home page doesn't have any important special state. (the other two do though, we still need to be able to switch to the home page and back)

You forgot the word "yet". Oh that nasty word.

Reply:
@somebody1234 there are still much better solutions than that.